dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -1446,6 +1440,45 @@ public boolean hasStartEndDate()
         return startDate != null && endDate != null;
     }
 
+    /**
+     * Indicates whether this query has a nonempty dateRangeList.
+     */
+    public boolean hasDateRangeList()
+    {
+        return dateRangeList != null && dateRangeList.size() > 0;
+    }
+
+    /**
+     * Indicates whether this query has a continuous or EMPTY dateRangeList.
+     */
+    public boolean hasContinuousDateRangeList()
+    {
+        // range date should be sorted in
+        // EventQueryParam::replacePeriodsWithDates method
+
+        if ( !hasDateRangeList() )
+        {
+            return true;
+        }
+
+        if ( dateRangeList.size() == 1 )
+        {
+            return true;
+        }
+
+        for ( int i = dateRangeList.size() - 1; i >= 0; i-- )
+        {
+            if ( i > 0 &&
```

Comment:
Do you think we can remove `i > 0`, and change the loop condition to `i > 0`, instead of `i >= 0`?